enumerables.rb

```diff
@@ -0,0 +1,145 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index(given_index = nil)
+    return enum_for(:my_each_with_index) unless block_given?
+
+    index = 0
+    index = given_index unless given_index.nil?
+    my_each do |x|
+      yield(x, index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    select_array = []
+    return enum_for(:my_select) unless block_given?
+
+    my_each do |x|
+      select_array << x if yield x
+    end
+    select_array
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |x|
+      return false if reg_exp(pattern, x) || num_exp(pattern, x)
+
+      unless block_given?
+        return false unless x
+
+        return true
+      end
+
+      return false unless yield x
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    my_each do |x|
+      return true if reg_exp(pattern, x) || num_exp(pattern, x)
+
+      unless block_given?
+        return true if x
+      end
+      return true if yield x
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    my_each do |x|
+      return false if reg_exp(pattern, x) || num_exp(pattern, x)
+
+      unless block_given?
+        return false if x
+
+        return true
+      end
+      return false if yield x
+    end
+    true
+  end
+
+  def my_count(items = nil)
+    repetitions = 0
+    my_each do |x|
+      unless pattern.nil?
+        return repetitions += 1 if items == x
+
+        return repetitions
+      end
+      unless block_given?
+        return repetitions += 1 if (yield x) == x
+
+        return repetitions
+      end
+    end
+    repetitions += 1 while repetitions < size
+    repetitions
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns the number of items in enum through enumeration
- [ ] counts the number of items in enum that are equal to item if an argument is given
- [ ] counts the number of elements yielding a true value if a block is given

Sample test case:
```ruby
let(:array) { Array.new(100) { rand(0...9) } }
expect(array.my_count).to eq array.count
```
Should return `true`